internal/testintegration/integration_test.go

```diff
@@ -178,26 +182,128 @@ func TestModifiesExistingDeployment(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Then we simulate the deployment pods being replaced
-	err = tcc.Client.Delete(ctx, rs1)
+	err = recreatePodsAfterDeploymentUpdate(ctx, tcc, d, rs1, pods)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// and check for 2 containers
+	err = tcc.ExpectPodContainerCount(ctx, d.Spec.Selector, 2, "all")
 	if err != nil {
 		t.Fatal(err)
 	}
+}
 
-	for i := 0; i < len(pods); i++ {
-		err = tcc.Client.Delete(ctx, pods[i])
-		if err != nil {
-			t.Fatal(err)
-		}
+func TestUpdateWorkloadContainerWhenDefaultProxyImageChanges(t *testing.T) {
```

Comment:
This is the test that demonstrates that when the operator's default image changes, it will automatically update the proxy container image on existing deployments.